js/messaging.js

```diff
@@ -232,65 +236,51 @@ function findExistTab({url, currentWindow, ignoreHash = true, ignoreSearch = fal
 /**
  * Opens a tab or activates an existing one,
  * reuses the New Tab page or about:blank if it's focused now
- * @param {Object} params
- *        or just a string e.g. openURL('foo')
- * @param {string} params.url
- *        if relative, it's auto-expanded to the full extension URL
- * @param {number} [params.index]
- *        move the tab to this index in the tab strip, -1 = last
- * @param {Boolean} [params.active]
- *        true to activate the tab (this is the default value in the extensions API),
- *        false to open in background
- * @param {?Boolean} [params.currentWindow]
- *        pass null to check all windows
- * @param {any} [params.message]
- *        JSONifiable data to be sent to the tab via sendMessage()
- * @returns {Promise<Tab>} Promise that resolves to the opened/activated tab
+ * @param {Object} _
+ * @param {string} _.url - if relative, it's auto-expanded to the full extension URL
+ * @param {number} [_.index] move the tab to this index in the tab strip, -1 = last
+ * @param {number} [_.openerTabId] defaults to the active tab
+ * @param {Boolean} [_.active=true] `true` to activate the tab
+ * @param {Boolean|null} [_.currentWindow=true] `null` to check all windows
+ * @param {Boolean} [_.newWindow=false] `true` to open a new window
+ * @param {chrome.windows.CreateData} [_.windowPosition] options for chrome.windows.create
+ * @returns {Promise<chrome.tabs.Tab>} Promise -> opened/activated tab
  */
-function openURL(options) {
-  if (typeof options === 'string') {
-    options = {url: options};
-  }
-  let {
-    url,
-    index,
-    active,
-    currentWindow = true,
-    newWindow = false,
-    windowPosition
-  } = options;
-
+function openURL({
+  url,
+  index,
+  openerTabId,
+  active = true,
+  currentWindow = true,
+  newWindow = false,
+  windowPosition,
+}) {
   if (!url.includes('://')) {
     url = chrome.runtime.getURL(url);
   }
-  return findExistTab({url, currentWindow}).then(tab => {
+  return findExistingTab({url, currentWindow}).then(tab => {
     if (tab) {
-      // update url if only hash is different?
-      // we can't update URL if !url.includes('#') since it refreshes the page
-      // FIXME: should we move the tab (i.e. specifying index)?
-      if (tab.url !== url && tab.url.split('#')[0] === url.split('#')[0] &&
-          url.includes('#')) {
-        return activateTab(tab, {url, index});
-      }
-      return activateTab(tab, {index});
-    }
-    if (newWindow) {
-      return createWindow(Object.assign({url}, windowPosition));
+      return activateTab(tab, {
+        index,
+        openerTabId,
+        // when hash is different we can only set `url` if it has # otherwise the tab would reload
+        url: url !== tab.url && url.includes('#') && url,
```

Comment:
Let's use a ternary:
```js
url !== tab.url && url.includes('#') ? url : undefined
```
I wonder if we can remove `url !== tab.url` check. Perhaps the tab reloads when the URL is the same.

Reply:
It should reload because it's like pressing the Enter key in the address bar.